seq2seq/nmt_without_attention_v2.py

```diff
@@ -0,0 +1,243 @@
+        # the encoded source sequence and the last generated target word.
+
+        # The encoded source sequence (encoder's output) must be specified by
+        # StaticInput, which is a read-only memory.
+        # Embedding of the last generated word is automatically gotten by
+        # GeneratedInputs, which is initialized by a start mark, such as <s>,
+        # and must be included in generation.
+
+        trg_embedding = paddle.layer.GeneratedInputV2(
+            size=target_dict_dim,
+            embedding_name='_target_language_embedding',
+            embedding_size=word_vector_dim)
+        group_inputs.append(trg_embedding)
+
+        beam_gen = paddle.layer.beam_search(
+            name=decoder_group_name,
+            step=gru_decoder_without_attention,
+            input=group_inputs,
+            bos_id=0,
+            eos_id=1,
+            beam_size=beam_size,
+            max_length=max_length)
+
+        return beam_gen
+
+
+def train(source_dict_dim, target_dict_dim):
+    cost = seq2seq_net(source_dict_dim, target_dict_dim)
+    parameters = paddle.parameters.create(cost)
+
+    # define optimize method and trainer
+    optimizer = paddle.optimizer.RMSProp(
+        learning_rate=1e-3,
+        gradient_clipping_threshold=10.0,
+        regularization=paddle.optimizer.L2Regularization(rate=8e-4))
+    trainer = paddle.trainer.SGD(
+        cost=cost, parameters=parameters, update_equation=optimizer)
+    # define data reader
+    wmt14_reader = paddle.batch(
+        paddle.reader.shuffle(
+            paddle.dataset.wmt14.train(source_dict_dim), buf_size=8192),
+        batch_size=55)
+
+    # define event_handler callback
+    def event_handler(event):
+        if isinstance(event, paddle.event.EndIteration):
+            if event.batch_id % 100 == 0 and event.batch_id > 0:
+                with gzip.open('models/nmt_without_att_params_batch_%d.tar.gz' %
+                               event.batch_id, 'w') as f:
+                    parameters.to_tar(f)
+
+            if event.batch_id % 10 == 0:
+                #                wmt14_test_batch = paddle.batch(
+                #                        paddle.reader.shuffle(
+                #                            paddle.dataset.wmt14.test(source_dict_dim), 
+                #                            buf_size=8192), batch_size=1)
+
+                #test_result = trainer.test(wmt14_test_batch)
+                print "\nPass %d, Batch %d, Cost %f, %s" % (
+                    event.pass_id,
+                    event.batch_id,
+                    event.cost,
+                    event.metrics,  # test_result.cost, test_result.metrics
+                )
+            else:
+                sys.stdout.write('.')
+                sys.stdout.flush()
+
+    # start to train
+    trainer.train(
+        reader=wmt14_reader, event_handler=event_handler, num_passes=2)
+
+
+def generate(source_dict_dim, target_dict_dim):
+    # use the first 3 samples for generation
```

Comment:
These codes are directly copied from PaddleBook which aim to demo that restricts sample number to generate, so that generation can finish in short time, but here we expect to provide an example that can directly be used by the users, the constraint should be removed.

After copying, please delete inappropriate codes and comments.